python/ray/dataframe/dataframe.py

```diff
@@ -22,29 +24,53 @@ def __init__(self, df, columns):
         assert(len(df) > 0)
 
         self._df = df
+        # TODO: Clean up later.
+        # We will call get only when we access the object (and only once).
+        self._lengths = \
+            ray.get([_deploy_func.remote(_get_lengths, d) for d in self._df])
```

Comment:
Another option (perhaps for later) that might make sense is to always pass along the length along with the object IDs, so `df` would be a list of `(Object_ID, length)` pairs. This assumes that anytime you create one of these partition object IDs, you can easily compute the length.

Reply:
I think we will probably need a way to either compute or set it. We will eventually have the exact same constructor as Pandas, so passing it won't be an option in the future.

I do think that most of the time we can easily compute the length on creation. I will think more about the optimization for this and we can review it later.